shell/platform/android/io/flutter/plugin/editing/TextEditingDelta.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/// A representation of the change that occured to an editing state, along with the resulting
+/// composing and selection regions.
+public final class TextEditingDelta {
+  private CharSequence oldText;
+  private CharSequence deltaText;
+  private int deltaStart;
+  private int deltaEnd;
+  private int newSelectionStart;
+  private int newSelectionEnd;
+  private int newComposingStart;
+  private int newComposingEnd;
+
+  private static final String TAG = "TextEditingDelta";
+
+  public TextEditingDelta(
+      CharSequence oldEditable,
+      int start,
```

Comment:
specify what kind of start/end. Since you have lots of similarly named vars, it is important to disambiguate whenever possible